### ISSUE TITLE
[feature] New attribute for highlight shortcode to allow span tag like inline but syntax highlighted code

### ISSUE THREAD

**Issue by kaushalmodi** [gohugoio/hugo#9442]
- [Discussion on Discourse](https://discourse.gohugo.io/t/is-there-a-way-to-inline-syntax-highlight-short-code-snippets/36834)
- [experiment](https://inline-highlight-sc-attempt--hugo-mwe.netlify.app/posts/inline-highlight-shortcode/#hack) | [markdown source](https://gitlab.com/hugo-mwe/hugo-mwe/-/raw/inline-highlight-sc-attempt/content/posts/inline-highlight-shortcode.md)

---

Right now, if I have this in the markdown:

```md
I am attempting to inline
{{< highlight emacs "linenos=inline" >}}(message "this highlight shortcode"){{< /highlight >}}.
But it is creating a paragraph break around it.
```

it renders to:

![image](https://user-images.githubusercontent.com/3578197/151622151-e3bf8e6a-345a-49a4-b7e6-80e12a87faa4.png)

This issue is to request a feature addition to the highlight shortcode with a boolean attribute like `inline`, where if `inline=true` is set, it will render `span` tags instead of `div` tags and remove the `pre` tags.

With a shortcode "hl4.html" defined this way:

```html
{{- transform.Highlight .Inner (.Get 0) (.Get 1 | default "")
    | replaceRE `^<div class="highlight"><pre [^>]+>((.|\n)+)</pre></div>$`
                `<span class="highlight-inline chroma">${1}</span>`
    | safeHTML -}}
```

and this in the markdown:

```md
I am attempting to inline
{{< hl4 emacs >}}(message "this hl4 shortcode"){{< /hl4>}}. And this!! finally works.
```

we can get this:

![image](https://user-images.githubusercontent.com/3578197/151622366-1a8aa9b7-adbc-4efb-9fb4-38a20e6cfdfe.png)

**Comment by kaushalmodi** [comment]
@bep I am following up on this request. 

I have started using this shortcode called `inline-src` in my posts.

## `inline-src.html` shortcode

```html
{{- transform.Highlight .Inner (.Get 0) (.Get 1 | default "")
    | replaceRE `^<div class="highlight"><pre [^>]+>((.|\n)+)</pre></div>$`
                `<span class="inline-src chroma">${1}</span>`
    | safeHTML -}}
```

.. and the results look great!

## Results

```md
If you liked how these _"generics"_ work in SystemVerilog and how the
{{< inline-src systemverilog >}}math #(int)::add(1, 2));{{< /inline-src >}} syntax
looks, check out the
```

![image](https://user-images.githubusercontent.com/3578197/153610577-e154c9bb-7e85-4742-bf35-8726c16ce051.png)

```md
In order to make `add` work with strings as well, we need this messy
_cast_ when the type name is _"string"_ : {{< inline-src systemverilog >}}$cast(ret_val, $sformatf("%s%s", a, b));{{< /inline-src >}}. It's basically
```

![image](https://user-images.githubusercontent.com/3578197/153611062-22447656-4b35-4ae8-97b0-427fe262a4d4.png)

## What's Next?

Can you include this shortcode as one of the inbuilt shortcodes?

**Comment by bep** [comment]
What I suggest we do to make this as consistent as possible without adding too much "new stuff":

* See https://github.com/gohugoio/hugo/blob/19f816f77e9c0a9caddfd112d525bad8b717da44/markup/highlight/highlight.go#L85
* If we add an option (e.g. `inline`) to the highlight shortcode + the regular hightlight template function
* Then we could change the signature on https://github.com/gohugoio/hugo/blob/19f816f77e9c0a9caddfd112d525bad8b717da44/markup/highlight/highlight.go#L71  to be inline with the other, and use the `.Inner` in the inline case.

**Comment by kaushalmodi** [comment]
Thanks. 

Adding an `inline` option to the `highlight` shortcode is good. 

I looked through the code snippets you linked but I have no clue how I would implement this. [My best go skill is adding a new test by copy-pasting and modifying an existing one :)].

Thanks for accepting this feature proposal. Hopefully someone can implement this.

**Comment by stefanv** [comment]
For others looking at the [GoodFirstIssue tag](https://github.com/gohugoio/hugo/labels/GoodFirstIssue), this one is now addressed by #9638 and is awaiting [Chroma#618](https://github.com/alecthomas/chroma/pull/618) which closes [Chroma#617](https://github.com/alecthomas/chroma/issues/617).
